Documentation/building/linux-instructions.md

```diff
@@ -1,14 +1,61 @@
 Build CoreCLR on Linux
 ======================
 
-This guide will walk you through building CoreCLR on Linux.  We'll start by showing how to set up your environment from scratch.
+This guide will walk you through building CoreCLR on Linux.  Before building there is environment setup that needs to happen to pull in all the dependencies required by the build.  There are two suggested ways to go about doing this. First you are able to use the docker environments provided by https://github.com/dotnet/dotnet-buildtools-prereqs-docker, or you can set up the environment yourself. The documentation will go over both ways of building; however, in general prefer docker if you are able. 
```

Comment:
I would not suggest preferring docker, but rather explain the benefits of each of the build ways. While it is a bit easier (you don't have to install all the prerequisites), the resulting binaries would most likely run on  the linux distro of the docker image only. 
When building without docker, you'd get binaries that work on your local distro whatever it is, but you need to make sure you have all the prerequisites installed.

It would be nice to suggest using the `-portable` option of build.sh that would allow the resulting binaries to work on distros with newer glibc version than the one the binaries were built on.

Reply:
I have removed this bit as I agree it is worth just presenting options. However, I am confused with the comment about only working on the environment you built in. I think in general it is unclear what we support for building/running on different oses. I do not believe that docker makes changes this, it just makes it easier to build/run on possibly different platforms.

That being said, I could easily today setup an environment to build on ubuntu and try to run those built binaries on debian and we provide what seems like little to no guidance that this should or should not work.

Reply:
I was referring to the fact that if you run your build in a docker container on e.g. Ubuntu 14.04 and use Ubuntu 16.04 docker image, then the binaries you've built won't work on your local machine, but only on Ubuntu 16.04 (and higher if you used the `-portable` option). People may expect that when you build using docker, the docker image is just a tool to build binaries that they can then run on the system where they ran the docker command.

Reply:
I see, I will add a warning.

Reply:
Have included this information. In https://github.com/dotnet/coreclr/pull/21967 we can add updated info for the portable build changes.